lib/src/components/json_widget_registry.dart

```diff
@@ -61,9 +61,12 @@ class JsonWidgetRegistry {
         _parent = parent {
     _logger = Logger('REGISTRY ${this.debugLabel}');
     _customBuilders.addAll(builders ?? {});
-    _internalFunctions =
-        internalFunctions ?? JsonWidgetInternalFunctions.defaults();
-    _functions.addAll(functions ?? {});
+    _functions.addAll({
+      ...overrideInternalFunctions
```

Comment:
This can actually be simplified slightly by using the `if` notation.
```
_functions.addAll({
  if (!overrideInternalFunctions) ...JsonWidgetInternalFunctions.defaults(),
  if (functions != null) ...functions,
});
```

Keeps two objects from being unnecessarily created.